src/main/java/de/torui/coflsky/handlers/DescriptionHandler.java

```diff
@@ -0,0 +1,177 @@
+            }
+        } else {
+            String itemId = ExtractStackableIdFromItemStack(itemStack);
+            if(tooltipItemIdMap.containsKey(itemId)){
+                return tooltipItemIdMap.getOrDefault(itemId, EMPTY_ARRAY);
+            }
+        }
+
+        return EMPTY_ARRAY;
+    }
+    public static void getTooltipDataFromBackend(GuiOpenEvent event){
+        try {
+            // delay a bit to wait for all inventory packages to arrive (each slot is sent individually)
+            Thread.sleep(100);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+        InventoryWrapper wrapper = new InventoryWrapper();
+
+        GuiContainer gc = (GuiContainer) event.gui;
+
+        if (event.gui instanceof GuiChest) {
+            ContainerChest chest = (ContainerChest) ((GuiChest) event.gui).inventorySlots;
+
+            IInventory inv = chest.getLowerChestInventory();
+            if (inv.hasCustomName()) {
+                String chestName = inv.getName();
+                wrapper.chestName = chestName;
+            }
+        }
+
+        NBTTagCompound compound = new NBTTagCompound();
+        NBTTagList tl = new NBTTagList();
+
+        for (Slot obj : gc.inventorySlots.inventorySlots) {
+            ItemStack stack = obj.getStack();
+            if (stack != null) {
+                tl.appendTag(stack.serializeNBT());
+            } else {
+                tl.appendTag(EMPTY_COMPOUND);
+            }
+        }
+
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        try {
+            compound.setTag("i", tl);
+            CompressedStreamTools.writeCompressed(compound, baos);
+
+            wrapper.fullInventoryNbt = Base64.getEncoder().encodeToString(baos.toByteArray());
+
+            String data = WSClient.gson.toJson(wrapper);
+            String info = QueryServerCommands.PostRequest("https://sky.coflnet.com/api/mod/description/modifications", data);
+
+            DescModification[][] arr = WSClient.gson.fromJson(info, DescModification[][].class);
+            int i = 0;
+            for (Slot obj : gc.inventorySlots.inventorySlots) {
+                ItemStack stack = obj.getStack();
+                if (stack == null) continue;
+                tooltipItemMap.put(stack, arr[i]);
+                String uuid = ExtractUuidFromItemStack(stack);
+                if(uuid.length()>0) tooltipItemUuidMap.put(uuid, arr[i]);
+
+                String id = ExtractStackableIdFromItemStack(stack);
+                if(id.length()>0) tooltipItemIdMap.put(id, arr[i]);
+                i++;
+            }
+
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+    public static void setTooltips(ItemTooltipEvent event) {
+        DescModification[] data = getTooltipData(event.itemStack);
+
+        if (data == null || data.length == 0)
+            return;
+
+        for (DescModification datum : data) {
+            switch (datum.type) {
```

Comment:
check if event.toolTip.size() is big enough to take the modification (bigger than datum.line)